subsys/bluetooth/audio/ascs.c

```diff
@@ -0,0 +1,1953 @@
+	 * ASE. The server shall send a notification of the ASE Control Point
+	 * characteristic to the client, and the server shall set the
+	 * Response_Code value for that ASE to 0x05 (Invalid ASE direction).
+	 */
+	if (ASE_DIR(ep->status.id) == BT_AUDIO_SINK) {
+		BT_ERR("Start failed: invalid operation for Sink");
+		ascs_cp_rsp_add(ASE_ID(ase), BT_ASCS_START_OP,
+				BT_ASCS_RSP_INVALID_DIR, BT_ASCS_REASON_NONE);
+		return;
+	}
+
+	stream = ep->stream;
+	if (unicast_server_cb != NULL && unicast_server_cb->start != NULL) {
+		err = unicast_server_cb->start(stream);
+	} else {
+		err = -EOPNOTSUPP;
+	}
+
+	if (err) {
+		BT_ERR("Start failed: %d", err);
+		ascs_cp_rsp_add(ASE_ID(ase), BT_ASCS_START_OP, err,
+				BT_ASCS_REASON_NONE);
+		return;
+	}
+
+	ascs_ep_set_state(ep, BT_AUDIO_EP_STATE_STREAMING);
+
+	ascs_cp_rsp_success(ASE_ID(ase), BT_ASCS_START_OP);
+}
+
+static ssize_t ascs_start(struct bt_ascs *ascs, struct net_buf_simple *buf)
+{
+	const struct bt_ascs_start_op *req;
+	int i;
+
+	if (buf->len < sizeof(*req)) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+	}
+
+	req = net_buf_simple_pull_mem(buf, sizeof(*req));
+
+	BT_DBG("num_ases %u", req->num_ases);
+
+	if (buf->len < req->num_ases) {
+		BT_ERR("Malformed ASE Start: len %u < %u", buf->len,
+		       req->num_ases);
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+	}
+
+	for (i = 0; i < req->num_ases; i++) {
+		struct bt_ascs_ase *ase;
+		uint8_t id;
+
+		id = net_buf_simple_pull_u8(buf);
+
+		BT_DBG("ase 0x%02x", id);
+
+		ase = ase_find(ascs, id);
+		if (!ase) {
+			ascs_cp_rsp_add(id, BT_ASCS_START_OP,
+					BT_ASCS_RSP_INVALID_ASE, 0x00);
+			BT_ERR("Unable to find ASE");
+			continue;
+		}
+
+		ase_start(ase);
+	}
+
+	return buf->size;
+}
+
+static ssize_t ascs_disable(struct bt_ascs *ascs, struct net_buf_simple *buf)
+{
+	const struct bt_ascs_disable_op *req;
+	int i;
+
+	if (buf->len < sizeof(*req)) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+	}
+
+	req = net_buf_simple_pull_mem(buf, sizeof(*req));
+
+	BT_DBG("num_ases %u", req->num_ases);
+
+	if (buf->len < req->num_ases) {
+		BT_ERR("Malformed ASE Disable: len %u < %u", buf->len,
+		       req->num_ases);
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+	}
+
+	for (i = 0; i < req->num_ases; i++) {
+		struct bt_ascs_ase *ase;
+		uint8_t id;
+
+		id = net_buf_simple_pull_u8(buf);
+
+		BT_DBG("ase 0x%02x", id);
+
+		ase = ase_find(ascs, id);
+		if (!ase) {
+			ascs_cp_rsp_add(id, BT_ASCS_DISABLE_OP,
+					BT_ASCS_RSP_INVALID_ASE_STATE, 0);
+			BT_ERR("Unable to find ASE");
+			continue;
+		}
+
+		ase_disable(ase);
+	}
+
+	return buf->size;
+}
+
+static void ase_stop(struct bt_ascs_ase *ase)
+{
+	struct bt_audio_stream *stream;
+	struct bt_audio_ep *ep;
+	int err;
+
+	BT_DBG("ase %p", ase);
+
+	ep = &ase->ep;
+
+	/* If the ASE_ID  written by the client represents a Sink ASE, the
+	 * server shall not accept the Receiver Start Ready operation for that
```

Comment:
This is the "stop" operation, not the "receiver start ready".
Seems like a copy and paste error.

Reply:
Seems like it, indeed. Updated to `stop`.